exporter.go

```diff
@@ -366,6 +398,35 @@ func (b *Exporter) handleEvent(event Event) {
 			conflictingEventStats.WithLabelValues("counter").Inc()
 		}
 
+	case *FlushEvent:
+		log.Debugf("FlushEvent")
+		for k, v := range b.Sets.store {
+			setevent := b.Sets.events[k]
+
+			b.handleEvent(&GaugeEvent{
+				relative:   false,
+				metricName: setevent.MetricName(),
+				value:      float64(len(v)),
+				labels:     setevent.labels,
+			})
+
+			b.Sets.store[k] = map[float64]bool{}
+		}
+
+	case *SetEvent:
+		store, found := b.Sets.store[metricName]
+		if found {
+			_, have := store[event.Value()]
+			if !have {
```

Comment:
it's customary in Go to collapse these lines to

```
if _, ok := store[event.Value()]; !ok {
	store[event.Value()] = true
}
```

so that it can go out of scope when no longer needed.

Reply:
I also see the variable usually called `ok` – you're calling it both `have` and `found`, I'd rather be consistent with those where possible.

Reply:
Did it that way to avoid aliasing.   Is aliasing generally not frowned on in Go code?

Reply:
it's probably better not to – although my instinct is that if you run into aliasing, you're nesting the code too deeply. One thing that took me a while to get used to in Go style is that we prefer

```
if … {
  … do stuff …
  return
}
… do other stuff …
```

rather than using an `else`. In most cases this means there is only one "ok" to work with. I don't see how to do that neatly here though, so I'm okay with leaving the names as they are.